src/main/java/net/glowstone/block/blocktype/BlockLamp.java

```diff
@@ -0,0 +1,34 @@
+package net.glowstone.block.blocktype;
+
+import net.glowstone.block.GlowBlock;
+import net.glowstone.block.GlowBlockState;
+import net.glowstone.entity.GlowPlayer;
+import org.bukkit.Material;
+import org.bukkit.block.BlockFace;
+import org.bukkit.inventory.ItemStack;
+
+/**
+ *
+ * @author Sam
+ */
+public class BlockLamp extends BlockType {
+
+    @Override
+    public void afterPlace(GlowPlayer player, GlowBlock block, ItemStack holding, GlowBlockState oldState) {
+        updatePhysics(block);
+    }
+
+    @Override
+    public void onNearBlockChanged(GlowBlock block, BlockFace face, GlowBlock changedBlock, Material oldType, byte oldData, Material newType, byte newData) {
+        updatePhysics(block);
+    }
+
+    @Override
+    public void updatePhysics(GlowBlock me) {
```

Comment:
There are moments in ticking and stuff where it calls updateBlock. I guess you could add updateBlock, and have it call updatePhysics